File: HackerRank/Algorithm/Strings/pangram/pangrams.pl
#!/usr/bin/perl
use strict;
use warnings;
use Data::Dumper;

# https://www.hackerrank.com/challenges/pangrams

# Head ends here

my $hash = {};

sub pangram {
    my $string = shift;
    foreach my $s (split('',$string)) {
        if ($s =~ /\w/i) {
            $hash->{lc $s} = 1;
        }

        if (scalar keys %{$hash} == 26) {
            print "pangram\n";
            return;
        }
    }
    print "not pangram\n";
    return;
}

#Tail starts here
my $input = <>;
chomp $input;
pangram($input);

File: HackerRank/Algorithm/Strings/pangram/pangrams.rb
#!/usr/bin/ruby


def pangram? (string)
    hash = Hash.new
    string.downcase.gsub(/\W/,'').split('').each { |s|
        unless hash.has_key?(s)
            hash[s] = 1
        end

        if hash.keys.size() == 26
            return true;
        end
    }
    return false;
end

puts pangram?(gets.chomp) ? "pangram" : "not pangram"
